pkg/build/api/types.go

```diff
@@ -425,6 +428,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
```

Comment:
the other option is we just go with a JenkinsStrategy that supports any Jenkins build; and uses a JenkinsSource which is the XML file configuration; that supports pipeline builds and any other kind of Jenkins build?

On the plus side this is much simpler on the OpenShift side (one String value in the BuildConfig source thats a blob of XML that can support any Jenkins build configuration of any kind). On the downside there's no pipeline abstraction (thats more an implementation detail deep in the XML blob) and the OOTB examples using Pipelines will look more verbose (since the Jenkinsfile gets encoded in Jenkins XML)